### PR TITLE
[TAS-98] Metadata

### PR DESCRIPTION
**Pull request by shamanth-kuthpadi**
**PR Requirements** (checkmark the following todo if applies)
- [ ] Tests were written to verify its correctness
- [ ] Document any code on Notion that requires documentation
- [ ] Run linting and prettier using `npm run lint && npm run format`
- [ ] Include picture(s) for any UI change (frontend only)

**PR Description**

### CODE REVIEW

File: backend/src/api/services/question/question.service.ts [hunk 1]
```diff
@@ -41,8 +44,33 @@ export const getSingleQuestionById = async (questionID: string): Promise<AnyQues
...
+    const chapters = await ChapterModel.find();
+    const metadata: SideBarMetadata[] = [];
+    for (const chapter of chapters) {
+        const questions = await QuestionBaseModel.find({ chapter: chapter.questions});
+        for (const question of questions) {
+            const student = await StudentProgressModel.findOne({ studentID });
+
+            if (!student) {
+                throw new HttpError(status.NOT_FOUND, `Student not found with ID ${studentID}`);
+            }
```
Comment (algebra2boy) [inline backend/src/api/services/question/question.service.ts:56]: I feel like this should be outside the for loop. This essentially tries to find a student for every question of every chapter. This logic should be performed once before traversing

**Review by algebra2boy — CHANGES_REQUESTED** [review]

File: backend/src/api/services/question/chapter.model.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import mongoose, { Document } from 'mongoose';
+import { chapter } from '../../validations/schemaLibrary.validation.js';
```
Comment (algebra2boy) [inline backend/src/api/services/question/chapter.model.ts:2]: I am just curious if you used this `chapter`.

Comment (Shamanth-GiT) [reply backend/src/api/services/question/chapter.model.ts:2]: I don't think I did.

File: backend/src/api/services/question/question.service.ts (hunk 1 above)
Comment (Shamanth-GiT) [reply backend/src/api/services/question/question.service.ts:56]: you are right

**Review by algebra2boy — APPROVED** [review]
Looks good. Thanks for doing this.

**Pull request merged**